packages/lesswrong/server/attributeEdits.ts

```diff
@@ -12,55 +12,75 @@ import fs from 'node:fs';
 type EditAttributions = (string|null)[]
 type InsDelUnc = "ins"|"del"|"unchanged"
 
-export async function annotateAuthors(documentId: string, collectionName: string, fieldName: string, upToVersion?: string|null): Promise<string> {
+export async function annotateAuthors(
+  documentId: string,
+  collectionName: string,
+  fieldName: string,
+  upToVersion?: string | null
+): Promise<string> {
+  const { finalHtml, attributions } = await computeAttributions(
+    documentId,
+    collectionName,
+    fieldName,
+    upToVersion
+  );
+
+  return attributionsToSpans(finalHtml, attributions);
+}
+
+export async function computeAttributions(
+  documentId: string,
+  collectionName: string,
+  fieldName: string,
+  upToVersion?: string | null
+): Promise<{ finalHtml: string; attributions: EditAttributions }> {
   const revs = await Revisions.find({
-    documentId, collectionName, fieldName
+    documentId,
+    collectionName,
+    fieldName,
   }).fetch();
-  if (!revs.length) return "";
-  
-  let filteredRevs = orderBy(revs, r=>r.editedAt);
-  
+  if (!revs.length) return { finalHtml: "", attributions: [] };
+
+  let filteredRevs = orderBy(revs, (r) => r.editedAt);
```

Comment:
As much as I think this style is better than the previous style, I think it's probably not worth the git blame, so would prefer to revert all the non-functional changes (and if the changes were made by an LLM I trust them even less and don't want to deal with the cognitive overhead of actually checking them)

Reply:
(for most of the raw diffs in this function)